src/renderer/atlas/AtlasEngine.h

```diff
@@ -173,6 +180,10 @@ namespace Microsoft::Console::Render::Atlas
             // UpdateHyperlinkHoveredId()
             u16 hyperlinkHoveredId = 0;
 
+            // These tracks the highlighted regions on the screen that are yet to be painted.
+            std::span<const til::point_span> searchHighlights;

```

Comment:
what keeps these pointing to valid memory?

Reply:
My understanding was that these are only used to smuggle data between different `IRenderEngine` API calls only within a single render pass (since the interface is stateful and AtlasEngine is more "snapshot-y").

Reply:
`_api.searchHighlights` is always reset with the new value at the start of each render pass. The highlights are read and drawn all under the lock, and if I'm not wrong, we never reset highlights without holding a lock on Terminal.

I do think that we now need to be extra  careful to always call `ResetIfStale()` under the lock, otherwise it could corrupt the AtlasEngine `_api.searchHighlights`. But, as @lhecker mentioned, in the future, the highlights will be stored as `std::vector` instead of `std::span` so there won't be any safety issues here.

Reply:
I think I should've atleast reset the spans to be empty in `EndPaint()`.